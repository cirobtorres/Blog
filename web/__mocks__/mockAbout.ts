import { faker } from "@faker-js/faker";
import { randomInt } from "./utilities/randomInt";
import { generateRandomCreatedAt } from "./utilities/randomISOdatetime";

export const createAboutMock = () => ({
  documentId: faker.string.uuid(),
  title: faker.lorem.words(randomInt(5, 12)),
  ...(() => {
    const github_link = faker.internet.url();
    const github_blog_link = github_link + "/" + faker.lorem.slug(3);
    return { github_link, github_blog_link };
  })(),
  ...generateRandomCreatedAt(),
  blocks: [],
});
