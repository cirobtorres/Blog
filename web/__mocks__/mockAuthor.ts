import { faker } from "@faker-js/faker";
import { randomInt } from "./utilities/randomInt";
import { randomRelativeURL } from "./utilities/randomRelativeURL";

export const createAuthorMock = (avatar = true) => ({
  documentId: faker.string.uuid(),
  name: faker.person.firstName(),
  avatar: avatar
    ? {
        documentId: faker.string.uuid(),
        ...randomRelativeURL(),
        alternativeText: faker.lorem.sentence(randomInt(5, 10)),
      }
    : null,
});
