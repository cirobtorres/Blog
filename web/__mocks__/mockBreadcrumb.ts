import { faker } from "@faker-js/faker";
import { randomInt } from "./utilities/randomInt";
import slugify from "@/utils/slugify";

export const createBreadcrumbMock = () => ({
  documentId: faker.string.uuid(),
  ...(() => {
    const name = faker.lorem.words(randomInt(2, 5));
    const slug = slugify(name);
    return { name, slug };
  })(),
  description: faker.lorem.sentence(randomInt(5, 12)),
});
