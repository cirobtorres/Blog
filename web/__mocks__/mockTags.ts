import { faker } from "@faker-js/faker";
import { randomInt } from "./utilities/randomInt";

export const createTagMock = () => ({
  documentId: faker.string.uuid(),
  name: faker.lorem.sentence(randomInt(1, 4)),
});

export const createTagsMock = () =>
  Array.from({ length: randomInt(5, 15) }, () => createTagMock());
