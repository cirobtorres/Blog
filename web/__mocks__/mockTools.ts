import { faker } from "@faker-js/faker";
import { randomInt } from "./utilities/randomInt";
import slugify from "../src/utils/slugify";

export const createToolMock = () => ({
  documentId: faker.string.uuid(),
  ...(() => {
    const name = faker.lorem.sentence(randomInt(5, 12));
    const slug = slugify(name);
    return { name, slug };
  })(),
  link: faker.system.directoryPath() + faker.system.commonFileName("png"),
  file: {
    documentId: faker.string.uuid(),
    url: faker.system.directoryPath() + faker.system.commonFileName("png"),
    alternativeText: faker.lorem.sentence(randomInt(5, 15)),
    caption: faker.lorem.sentence(randomInt(9, 20)),
    width: 30,
    height: 30,
  },
});

export const createToolsMock = () =>
  Array.from({ length: randomInt(5, 15) }, () => createToolMock());
