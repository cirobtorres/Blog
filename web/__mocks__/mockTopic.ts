import { faker } from "@faker-js/faker";

export const mockTopic = () => ({
  documentId: faker.string.uuid(),
  name: faker.lorem.word(),
  slug: faker.lorem.slug(),
  description: faker.lorem.sentence(),
});
