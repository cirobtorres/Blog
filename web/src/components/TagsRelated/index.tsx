import Link from "next/link";
import {
  Tooltip,
  TooltipContent,
  TooltipProvider,
  TooltipTrigger,
} from "../Shadcnui/tooltip";
import { DotBackground } from "../Backgrounds";
import Image from "next/image";

const TagsRelated = ({
  topic,
  tools,
  tags,
}: {
  topic: Topic;
  tools?: Tool[];
  tags?: Tag[];
}) => {
  return (
    <section className="mb-10">
      <DotBackground className="relative grid grid-rows-[repeat(3,minmax(20px,1fr))] justify-center pt-20 pb-[81px]">
        <div className="absolute top-0 h-20 left-0 right-0 opacity-90 bg-blog-fade-dot-up" />
        {topic && <Topic topic={topic} />}
        {tools && tools.length > 0 && <Tools tools={tools} />}
        {tags && tags.length > 0 && <Tags tags={tags} />}
        <div className="absolute bottom-0 h-20 left-0 right-0 opacity-90 bg-blog-fade-dot-down" />
      </DotBackground>
    </section>
  );
};

const Topic = ({ topic }: { topic: Topic }) => {
  return (
    <article
      className="max-w-screen-2xl mx-auto flex items-center"
      role="region"
      data-testid="article-topic"
      aria-label={`TÃ³pico: ${topic.name}`}
    >
      <div className="flex justify-center flex-wrap gap-6 col-start-2 max-lg:col-start-1 px-4">
        <div className="w-full flex justify-center">
          <h2
            id={`topic-title-${topic.documentId}`}
            className="text-[2.5rem] font-extrabold text-center max-[800px]:text-[2rem]"
          >
            {topic.name}
          </h2>
        </div>
      </div>
    </article>
  );
};

const Tools = ({ tools }: { tools: Tool[] }) => {
  return (
    <article
      data-testid="article-tools"
      aria-label="Ferramentas e tecnologias"
      className="max-w-screen-2xl mx-auto flex items-center"
    >
      <ul
        data-testid="article-tools-list"
        role="list"
        aria-label="Lista das ferramentas e tecnologias abordadas neste artigo"
        className="max-w-[500px] mx-auto flex flex-wrap gap-6 justify-center items-center px-4"
      >
        {tools.map((tool) => (
          <li
            key={tool.documentId}
            data-testid={tool.documentId}
            role="listitem"
            className="flex items-center gap-2"
          >
            <Image
              src={
                tool.file
                  ? process.env.NEXT_PUBLIC_BACKEND_IP + tool.file.url
                  : "https://placehold.co/32x32/171717/FFFFFF/png"
              }
              alt={
                tool.file ? tool.file.alternativeText : `Logo do ${tool.name}`
              }
              width={32}
              height={32}
              // className="rounded-full border-2 border-blog-border bg-black"
            />
            <TooltipProvider>
              <Tooltip>
                <TooltipTrigger asChild>
                  <Link
                    href={tool.link}
                    target="_blank"
                    rel="noopener noreferrer"
                    aria-label={`Abrir nova aba com site oficial de ${tool.name}`}
                    className="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
                  >
                    {tool.name}
                  </Link>
                </TooltipTrigger>
                <TooltipContent>
                  <p>{tool.link}</p>
                </TooltipContent>
              </Tooltip>
            </TooltipProvider>
          </li>
        ))}
      </ul>
    </article>
  );
};

const Tags = ({ tags }: { tags: Tag[] }) => {
  return (
    <article data-testid="article-tags" aria-label="Tags" className="">
      <div className="max-w-screen-lg mx-auto">
        <ul
          role="list"
          data-testid="article-tags-list"
          aria-label="Lista de tags"
          className="flex justify-center flex-wrap gap-x-4 gap-y-2 blog-heading col-start-2 max-lg:col-start-1 px-40 max-[800px]:px-4"
        >
          {tags.map((tag, index) => (
            <li
              data-testid={tag.documentId}
              key={tag.documentId}
              role="listitem"
              aria-labelledby={`Tag-${index + 1}`}
              className="flex items-center gap-2"
            >
              <span className="text-[#808080] italic">{tag.name}</span>
            </li>
          ))}
        </ul>
      </div>
    </article>
  );
};

export default TagsRelated;
