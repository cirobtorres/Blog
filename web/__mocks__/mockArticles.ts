import { faker } from "@faker-js/faker";
import { randomInt } from "./utilities/randomInt";
import { generateRandomCreatedAt } from "./utilities/randomISOdatetime";
import slugify from "@/utils/slugify";
import { createCoverMock } from "./mockCover";
import { createAuthorMock } from "./mockAuthor";
import { mockTopic } from "./mockTopic";

export function createArticleMock() {
  return {
    documentId: faker.string.uuid(),
    ...(() => {
      const title = faker.lorem.sentence(randomInt(5, 12));
      const slug = slugify(title);
      return { title, slug };
    })(),
    description: faker.lorem.sentence(randomInt(14, 28)),
    ...generateRandomCreatedAt(),
    cover: createCoverMock(),
    author: createAuthorMock(),
    topic: mockTopic(),
    tools: [],
    tags: [],
    blocks: [],
  };
}

export const mockArticles = (length = randomInt(5, 10)) =>
  Array.from({ length }, () => ({
    ...createArticleMock(),
  }));

export const mockArticle = createArticleMock();

export const mockArticlesCount = {
  articles_connection: {
    pageInfo: { total: 100, pageCount: 10, pageSize: 10, page: 1 },
  },
};
