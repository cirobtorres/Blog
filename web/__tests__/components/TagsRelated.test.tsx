import TagsRelated from "@/components/TagsRelated";
import { render, screen } from "@testing-library/react";
import { mockTopic } from "../../__mocks__/mockTopic";
import { createToolsMock } from "../../__mocks__/mockTools";
import { createTagsMock } from "../../__mocks__/mockTags";
import { faker } from "@faker-js/faker";

faker.seed(4); // Snapshot

describe("Categories", () => {
  const topicMocked = mockTopic();

  const toolsMocked = createToolsMock();

  const tagsMocked = createTagsMock();

  beforeEach(() => {
    process.env.NEXT_PUBLIC_BACKEND_IP = "http://127.0.0.1:1337";
  });

  test("Categories component matches the snapshot", () => {
    const { asFragment } = render(
      <TagsRelated topic={topicMocked} tools={toolsMocked} tags={tagsMocked} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe("Topic", () => {
    test("Topic to be in the document", () => {
      render(<TagsRelated topic={topicMocked} />);

      const topic = screen.queryByTestId("article-topic");
      expect(topic).toBeInTheDocument();
    });

    test("Topic accessibility attributes are rendered", () => {
      render(<TagsRelated topic={topicMocked} />);

      const topic = screen.queryByTestId("article-topic");
      expect(topic).toHaveAttribute("role", "region");
      expect(topic).toHaveAttribute(
        "aria-label",
        `TÃ³pico: ${topicMocked.name}`
      );
      const h2Component = topic?.querySelector("h2");
      expect(h2Component).toHaveAttribute(
        "id",
        `topic-title-${topicMocked.documentId}`
      );
      expect(h2Component).toHaveTextContent(topicMocked.name);
    });
  });

  describe("Tools", () => {
    test("Tools to not be in the document if article has no tools", () => {
      render(<TagsRelated topic={topicMocked} />);

      const tools = screen.queryByTestId("article-tools");
      expect(tools).toBeNull();
      expect(tools).not.toBeInTheDocument();
    });

    test("Subcategories to not be in the document if tools is an empty array", () => {
      render(<TagsRelated topic={topicMocked} tools={[]} />);

      const tools = screen.queryByTestId("article-tools");
      expect(tools).toBeNull();
      expect(tools).not.toBeInTheDocument();
    });

    test("Subcategories accessibility attributes are rendered", () => {
      render(<TagsRelated topic={topicMocked} tools={toolsMocked} />);

      // article
      const tools = screen.queryByTestId("article-tools");
      expect(tools).toHaveAttribute("aria-label", "Ferramentas e tecnologias");

      // ul
      const toolsList = screen.queryByTestId("article-tools-list");
      expect(toolsList).toHaveAttribute("role", "list");
      expect(toolsList).toHaveAttribute(
        "aria-label",
        "Lista das ferramentas e tecnologias abordadas neste artigo"
      );

      const listItems = toolsList?.getElementsByTagName("li");
      expect(listItems?.length).toBe(toolsMocked.length);

      // li
      Array.from(listItems || []).forEach((listItem, index) => {
        const tool = toolsMocked[index];
        expect(listItem).toHaveAttribute("data-testid", tool.documentId);
        expect(listItem).toHaveAttribute("role", "listitem");

        const linklement = listItem.querySelector("a");
        expect(linklement).toHaveAttribute("target", "_blank");
        expect(linklement).toHaveAttribute("rel", "noopener noreferrer");
        expect(linklement).toHaveClass(
          "text-blog-foreground-readable hover:text-blog-foreground-readable-hover"
        );
      });
    });
  });

  describe("Tags", () => {
    test("Tags to not be in the document if article has no tag", () => {
      render(<TagsRelated topic={topicMocked} tools={toolsMocked} />);

      const tags = screen.queryByTestId("article-tags");
      expect(tags).toBeNull();
      expect(tags).not.toBeInTheDocument();
    });

    test("Tags to not be in the document if article is an empty array", () => {
      render(<TagsRelated topic={topicMocked} tags={[]} />);

      const tags = screen.queryByTestId("article-tags");
      expect(tags).toBeNull();
      expect(tags).not.toBeInTheDocument();
    });

    test("Tags to be in the document if tags is a non empty array of tags", () => {
      render(<TagsRelated topic={topicMocked} tags={tagsMocked} />);

      const tags = screen.queryByTestId("article-tags");
      expect(tags).toBeInTheDocument();

      tagsMocked.forEach((tag) => {
        const listItem = screen.getByTestId(tag.documentId);
        expect(listItem).toBeInTheDocument();
      });
    });

    test("Tags accessibility attributes are rendered", () => {
      render(<TagsRelated topic={topicMocked} tags={tagsMocked} />);

      // article
      const tag = screen.queryByTestId("article-tags");
      expect(tag).toHaveAttribute("aria-label", "Tags");

      // ul
      const tagsList = screen.queryByTestId("article-tags-list");
      expect(tagsList).toHaveAttribute("role", "list");
      expect(tagsList).toHaveAttribute("aria-label", "Lista de tags");

      const listItems = tagsList?.getElementsByTagName("li");
      expect(listItems?.length).toBe(tagsMocked.length);

      // li
      Array.from(listItems || []).forEach((listItem, index) => {
        const tag = tagsMocked[index];

        expect(listItem).toHaveAttribute("data-testid", tag.documentId);
        expect(listItem).toHaveAttribute("role", "listitem");

        const spanElement = listItem.querySelector("span");
        expect(spanElement).toHaveClass("text-[#808080] italic");
      });
    });
  });
});
