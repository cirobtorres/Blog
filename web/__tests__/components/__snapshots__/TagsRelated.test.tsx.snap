// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Categories component matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="mb-10"
  >
    <div
      class="bg-fixed bg-[size:8px_8px] bg-[image:radial-gradient(hsl(0,_0%,_94%)_1px,_transparent_0)] dark:bg-[image:radial-gradient(hsl(0,_0%,_16%)_1px,_transparent_0)] after:absolute after:inset-0 after:pointer-events-none after:rounded-lg after:inset-ring after:inset-ring-gray-950/50 relative grid grid-rows-[repeat(3,minmax(20px,1fr))] justify-center pt-20 pb-[81px]"
    >
      <div
        class="absolute top-0 h-20 left-0 right-0 opacity-90 bg-blog-fade-dot-up"
      />
      <article
        aria-label="TÃ³pico: viduo"
        class="max-w-screen-2xl mx-auto flex items-center"
        data-testid="article-topic"
        role="region"
      >
        <div
          class="flex justify-center flex-wrap gap-6 col-start-2 max-lg:col-start-1 px-4"
        >
          <div
            class="w-full flex justify-center"
          >
            <h2
              class="text-[2.5rem] font-extrabold text-center max-[800px]:text-[2rem]"
              id="topic-title-f8fbb3f0-46c3-4df2-a906-0f6fcd2192b6"
            >
              viduo
            </h2>
          </div>
        </div>
      </article>
      <article
        aria-label="Ferramentas e tecnologias"
        class="max-w-screen-2xl mx-auto flex items-center"
        data-testid="article-tools"
      >
        <ul
          aria-label="Lista das ferramentas e tecnologias abordadas neste artigo"
          class="max-w-[500px] mx-auto flex flex-wrap gap-6 justify-center items-center px-4"
          data-testid="article-tools-list"
          role="list"
        >
          <li
            class="flex items-center gap-2"
            data-testid="68a489e5-a372-4e6d-86e6-59dd9193be89"
            role="listitem"
          >
            <img
              alt="Considero vespillo suppellex ipsam attollo cunae tenax vulariter aer appono taedium triumphus alienus."
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fopt%2Fsharelest.png&w=64&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fopt%2Fsharelest.png&w=32&q=75 1x, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fopt%2Fsharelest.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <a
              aria-label="Abrir nova aba com site oficial de Aperio porro convoco tenetur administratio amo stella dolorum mollitia."
              class="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
              data-state="closed"
              href="/usr/portsunnaturally_distinction.png"
              rel="noopener noreferrer"
              target="_blank"
            >
              Aperio porro convoco tenetur administratio amo stella dolorum mollitia.
            </a>
          </li>
          <li
            class="flex items-center gap-2"
            data-testid="f2354416-08f7-499e-9bd5-e26b5ed230d5"
            role="listitem"
          >
            <img
              alt="Dicta soleo vilicus titulus anser venustas spiculum comitatus sustineo."
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fhomeovercharge_procrastinate.png&w=64&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fhomeovercharge_procrastinate.png&w=32&q=75 1x, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fhomeovercharge_procrastinate.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <a
              aria-label="Abrir nova aba com site oficial de Anser defluo canonicus magnam balbus dolorem vinum."
              class="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
              data-state="closed"
              href="/opt/binscamper_delectable_chip.png"
              rel="noopener noreferrer"
              target="_blank"
            >
              Anser defluo canonicus magnam balbus dolorem vinum.
            </a>
          </li>
          <li
            class="flex items-center gap-2"
            data-testid="1c51b809-c0a4-4618-bf41-df7eebd2d648"
            role="listitem"
          >
            <img
              alt="Animus amita eaque soleo trucido molestias concido substantia audeo concedo."
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fprocstiff_oof.png&w=64&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fprocstiff_oof.png&w=32&q=75 1x, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fprocstiff_oof.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <a
              aria-label="Abrir nova aba com site oficial de Aetas cito circumvenio subiungo vis inventore catena succedo."
              class="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
              data-state="closed"
              href="/etc/namedbum.png"
              rel="noopener noreferrer"
              target="_blank"
            >
              Aetas cito circumvenio subiungo vis inventore catena succedo.
            </a>
          </li>
          <li
            class="flex items-center gap-2"
            data-testid="22c171fd-3db1-44ab-b527-c834c7c2ace4"
            role="listitem"
          >
            <img
              alt="Aegrus trucido tutamen vergo texo tubineus."
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Flost%2Bfoundgah_er.png&w=64&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Flost%2Bfoundgah_er.png&w=32&q=75 1x, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Flost%2Bfoundgah_er.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <a
              aria-label="Abrir nova aba com site oficial de Claudeo adaugeo sono vir natus coepi comprehendo deprecator vorago."
              class="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
              data-state="closed"
              href="/opt/sbinjacket_energetically.png"
              rel="noopener noreferrer"
              target="_blank"
            >
              Claudeo adaugeo sono vir natus coepi comprehendo deprecator vorago.
            </a>
          </li>
          <li
            class="flex items-center gap-2"
            data-testid="1a01ef13-fea7-4c7f-bca1-b12d99bdf101"
            role="listitem"
          >
            <img
              alt="Sui adipiscor delectatio corpus aequus nobis aetas stella desolo urbs."
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fetc%2Fmailcasement.png&w=64&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fetc%2Fmailcasement.png&w=32&q=75 1x, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fetc%2Fmailcasement.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <a
              aria-label="Abrir nova aba com site oficial de Coruscus peior animus creta clementia spero aro voluptate adhuc."
              class="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
              data-state="closed"
              href="/etc/defaultsreplacement.png"
              rel="noopener noreferrer"
              target="_blank"
            >
              Coruscus peior animus creta clementia spero aro voluptate adhuc.
            </a>
          </li>
          <li
            class="flex items-center gap-2"
            data-testid="655ad8b9-b92d-4248-bae2-099b2c12210a"
            role="listitem"
          >
            <img
              alt="Beatus atavus doloremque basium tamquam."
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fetc%2Fdefaultsunless_lady.png&w=64&q=75"
              srcset="/_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fetc%2Fdefaultsunless_lady.png&w=32&q=75 1x, /_next/image?url=http%3A%2F%2F127.0.0.1%3A1337%2Fetc%2Fdefaultsunless_lady.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <a
              aria-label="Abrir nova aba com site oficial de Argentum clementia caries sumo quasi perferendis vorax cetera."
              class="text-blog-foreground-readable hover:text-blog-foreground-readable-hover rounded focus-within:outline focus-within:outline-2 focus-within:outline-blog-foreground-readable-hover"
              data-state="closed"
              href="/Systemphew_luck.png"
              rel="noopener noreferrer"
              target="_blank"
            >
              Argentum clementia caries sumo quasi perferendis vorax cetera.
            </a>
          </li>
        </ul>
      </article>
      <article
        aria-label="Tags"
        class=""
        data-testid="article-tags"
      >
        <div
          class="max-w-screen-lg mx-auto"
        >
          <ul
            aria-label="Lista de tags"
            class="flex justify-center flex-wrap gap-x-4 gap-y-2 blog-heading col-start-2 max-lg:col-start-1 px-40 max-[800px]:px-4"
            data-testid="article-tags-list"
            role="list"
          >
            <li
              aria-labelledby="Tag-1"
              class="flex items-center gap-2"
              data-testid="b65ae4a6-0000-44cf-a6ce-4af9b165c2c8"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Tenetur fuga certus cribro.
              </span>
            </li>
            <li
              aria-labelledby="Tag-2"
              class="flex items-center gap-2"
              data-testid="34a6f3b4-e8ee-4fca-97e7-fb9452cab7f2"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Adhuc suppellex.
              </span>
            </li>
            <li
              aria-labelledby="Tag-3"
              class="flex items-center gap-2"
              data-testid="439d087f-51f7-4784-8ab9-96a1509feb8c"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Congregatio dedico.
              </span>
            </li>
            <li
              aria-labelledby="Tag-4"
              class="flex items-center gap-2"
              data-testid="ec8825ae-9bca-4dad-8eec-081d705cceab"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Amor patrocinor.
              </span>
            </li>
            <li
              aria-labelledby="Tag-5"
              class="flex items-center gap-2"
              data-testid="73bd2d4d-a87d-49cf-824a-22fee4870026"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Curo.
              </span>
            </li>
            <li
              aria-labelledby="Tag-6"
              class="flex items-center gap-2"
              data-testid="c095fbec-a160-4f92-b45a-4f4d8154f6a5"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Eius ustulo.
              </span>
            </li>
            <li
              aria-labelledby="Tag-7"
              class="flex items-center gap-2"
              data-testid="36a9af44-d70e-4e66-9f51-a42a1f35e90e"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Cattus.
              </span>
            </li>
            <li
              aria-labelledby="Tag-8"
              class="flex items-center gap-2"
              data-testid="8c87ce71-2efb-49ec-ab8b-5dc806729042"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Attonbitus.
              </span>
            </li>
            <li
              aria-labelledby="Tag-9"
              class="flex items-center gap-2"
              data-testid="f4830b57-028a-4af5-9dcb-deb0398f9ca1"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Arceo deleniti.
              </span>
            </li>
            <li
              aria-labelledby="Tag-10"
              class="flex items-center gap-2"
              data-testid="9cadd1e6-a447-4f6f-8fc6-6c61d4acbf8e"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Fuga officiis.
              </span>
            </li>
            <li
              aria-labelledby="Tag-11"
              class="flex items-center gap-2"
              data-testid="3cef5f64-b032-45e2-bfcd-7e9e5ff62f3c"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Sursum.
              </span>
            </li>
            <li
              aria-labelledby="Tag-12"
              class="flex items-center gap-2"
              data-testid="18d13cbc-845b-4f43-b512-d12d3f426068"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Creptio correptius cupressus.
              </span>
            </li>
            <li
              aria-labelledby="Tag-13"
              class="flex items-center gap-2"
              data-testid="322cf849-295c-4ff1-a33f-b2054df4c560"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Deprimo.
              </span>
            </li>
            <li
              aria-labelledby="Tag-14"
              class="flex items-center gap-2"
              data-testid="2d378623-81ed-4117-80f1-88968611e6ad"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Soleo.
              </span>
            </li>
            <li
              aria-labelledby="Tag-15"
              class="flex items-center gap-2"
              data-testid="4765cb27-ef6d-452b-9707-465b1867a88b"
              role="listitem"
            >
              <span
                class="text-[#808080] italic"
              >
                Stillicidium cometes repudiandae.
              </span>
            </li>
          </ul>
        </div>
      </article>
      <div
        class="absolute bottom-0 h-20 left-0 right-0 opacity-90 bg-blog-fade-dot-down"
      />
    </div>
  </section>
</DocumentFragment>
`;
